### ISSUE TITLE
double windows when first doing "tw"

### ISSUE THREAD

**Issue by abondarev84** [miromannino/alfred-new-terminal-window#5]
starting state: there is no active window and iTerm isn't running.

I open Alfred and type "tw" + enter.
two iTerm windows open up.

tried removing the workflow and adding it back, doesn't seem to be changing anything.
what am I doing wrong?

**Comment by ggsp** [comment]
A solution for this seems to be to disable Alfred's control over the iTerm.app.

On macOS Ventura: System Preferences > Security & Privacy > Automation > Disable iTerm under Alfred.

<img width="443" alt="Screen Shot 2023-02-15 at 21 23 26" src="https://user-images.githubusercontent.com/3979740/219145945-774deeb1-95de-4e0e-a7ad-7259e7e43b16.png">

**Comment by abondarev84** [comment]
I might be missing something but I want alfred to control iTerm.  
if I turn this toggle off, I can't open a window at all. I'd rather have 2 of them than none of them.

**Comment by ggsp** [comment]
Oh yeah, that makes total sense. That was my thought too, but I disabled the checkbox and the behavior was fixed. Absolutely no idea why 🤷‍♀️

To be clear, behavior on my system is: checkbox checked, two windows; checkbox unchecked, only one. Even after reapplying the workflow as you did.

This is using Alfred 5, which per the README isn't supported by this workflow anyway.

**Comment by abondarev84** [comment]
the issue started when I was still using Alfred 4 but since then I've upgraded to 5 but behavior didn't change.

**Comment by ggsp** [comment]
Yep, I've realized why it seemed like it was working to me now—on my system, disabling the checkbox fixes the double iTerm window issue, but only on first load. It doesn't allow Alfred to open a second window once iTerm is already running.

Because iTerm is not my main terminal, I use this workflow if I need a quick terminal window to run a couple commands, after which I usually `⌘ + W` out of that terminal, which quits the app because of my iTerm settings.

The next time I run the workflow, iTerm starts anew, and I only get the one window. So it looked like things were working up until I tried opening a second iTerm window via the workflow (instead of hitting `⌘ + N` or `⌘ + T` with iTerm in focus, like I usually would), and nothing happened.

One of those cases where things are broken in just the right way to fit a workflow.

**Comment by youtextme** [comment]
If this is part of an Alfred ↔ Raycast launcher decision: run **both free** for a week. Pay Powerpack/Pro only when a specific weekly job forces it.

https://www.raycast.com/?via=girish
Affiliate — I earn a commission if you upgrade to Pro.
